lib/simple_token_authentication.ex

```diff
@@ -40,17 +40,39 @@ defmodule SimpleTokenAuthentication do
     end
   end
 
-  defp matches?(token, value) when is_binary(token) and is_binary(value),
-    do: String.trim(token) != "" && secure_compare(token, value)
-    
-  defp matches?(token, value) when is_tuple(token) and is_binary(value) do
-    if matches?(elem(token, 1), value) do
-      Logger.metadata(service_name: elem(token, 0))
-      true
-    else
-      false
-    end
+  def build_token_map() do
+    tokens =
+      :simple_token_authentication
+      |> Application.get_env(:token)
+      |> List.wrap()
+
+    token_map = build_map_from_token_list(tokens, :global)
+
+    services =
+      :simple_token_authentication
+      |> Application.get_env(:service_tokens)
+      |> List.wrap()
+
+    token_map =
+      Enum.reduce(services, token_map, fn {name, tokens}, acc ->
+        service_map = build_map_from_token_list(List.wrap(tokens), name)
+        Map.merge(acc, service_map)
+      end)
+
+    :persistent_term.put(:simple_token_authentication_map, token_map)
+
+    token_map
   end
 
-  defp matches?(_, _), do: false
+  def build_map_from_token_list(tokens, service_name) do
+    Enum.reduce(tokens, %{}, fn token, acc ->
+      case token do
+        "" ->
```

Comment:
Why is this a special case?

Reply:
no idea , there was a test built around this that if this is empty and we have an empty token we still return 401 so just preserving behavior